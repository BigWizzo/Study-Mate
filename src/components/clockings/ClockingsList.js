import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { useParams, withRouter } from 'react-router-dom';
import { clockingFetchRequest } from '../../redux/clockings/clockingActions';
import ClockingListItem from './ClockingListItem';
import { dateShort } from '../../utils/utils';

const ClockingsList = ({
  loadClockings, clockings, history, student,
}) => {
  const { id } = useParams();

  useEffect(() => {
    if (!student) {
      history.push('/login');
    } else {
      loadClockings(id);
    }
  }, [id, loadClockings, history, student]);

  return (
    <div>
      {clockings
        && clockings.map((clocking) => (
          <ClockingListItem
            clocking={clocking}
            key={clocking?.id}
            topic={clocking?.topic}
            duration={clocking?.duration}
            created_at={dateShort(clocking?.created_at)}
          />
        ))}
      {clockings < 1 && (
        <h3 className="text-center Clockings my-4">No Clockings to show</h3>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({
  clockings: state.clockings.clockings,
  student: state.student.student,
});

const mapDispatchToProps = (dispatch) => ({
  loadClockings: (id) => dispatch(clockingFetchRequest(id)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withRouter(ClockingsList));
