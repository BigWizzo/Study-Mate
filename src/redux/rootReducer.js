import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import studentReducer from './students/studentReducer';
import subjectReducer from './subjects/subjectReducer';
import clockingReducer from './clockings/clockingReducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['student', 'subjects', 'clockings'],
};

const rootReducer = combineReducers({
  student: studentReducer,
  subjects: subjectReducer,
  clockings: clockingReducer,
});

export default persistReducer(persistConfig, rootReducer);
