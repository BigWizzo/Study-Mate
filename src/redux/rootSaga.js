import { all, call } from 'redux-saga/effects';
import { studentSagas } from './students/studentSagas';
import { subjectSagas } from './subjects/subjectSagas';

export default function* rootSaga() {
  yield all([call(studentSagas), call(subjectSagas)]);
}
