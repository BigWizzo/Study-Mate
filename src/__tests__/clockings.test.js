import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { store, persistor } from '../redux/store';
import ClockingsList from '../components/clockings/ClockingsList';

const clocking = {
  id: 1,
  topic: 'topic 1',
  details: 'details 1',
  subject_id: 1,
};

test('should render the clockings list component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <BrowserRouter>
          <ClockingsList
            id={clocking?.id}
            title={clocking?.topic}
            description={clocking?.details}
            clocking={clocking}
          />
        </BrowserRouter>
      </PersistGate>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
