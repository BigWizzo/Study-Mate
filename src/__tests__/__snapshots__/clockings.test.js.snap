// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the clockings list component 1`] = `
<div>
  <h3
    className="text-center Clockings my-4"
  >
    No Clockings to show
  </h3>
</div>
`;
